### ISSUE TITLE
Is there a reason this is using Neo and not OPT?

### ISSUE THREAD

**Issue by Delcos** [harubaru/convogpt#2]
Opt has beat out older models in almost all instances, I'm wondering if there is any particular reason to use Neo.

**Comment by harubaru** [comment]
It seems a bit controversial as to whether or not OPT beats Neo or Neo beats OPT, at least from some places I've asked. I've used Neo as the base for 125M, 1.3B, and 2.7B since people recommended to me that those were good choices over OPT. I'm planning on using OPT 13B and above though

**Comment by Delcos** [comment]
> It seems a bit controversial as to whether or not OPT beats Neo or Neo beats OPT, at least from some places I've asked. I've used Neo as the base for 125M, 1.3B, and 2.7B since people recommended to me that those were good choices over OPT. I'm planning on using OPT 13B and above though

That's a good idea. Is there a way to get a normal pytorch_model.bin instead of safetensor?

**Comment by harubaru** [comment]
I probably need a better way to host the weights, I keep running out of disk space when I've tried uploading the pickled weights

**Comment by scatterp2** [comment]
i was wondering last night and i may be wrong here but could the training data lets say twitter be encoded such that a base52 scheme is used what do i mean by this well lets take an example "word" which occurs approximately 92.31% of the way through an English dictionary which could contain up to 1.25M words including rare scientific words etc this means "word" in our lookup table is 1153875 base52 encoded 
Here is the complete base 52 encoding of 1153875:
1153875 / 52 = 22074 R 25
22074 / 52 = 423 R 24
423 / 52 = 8 R 15
8 / 52 = 0 R 8
Thus, the base 52 encoding of 1153875 is zzzy.

which saves nothing but if every word becomes 4 letters, you have a new language, basically 
symbolic representations for the bot to use for its thoughts 
not sure if this is any use or not?

**Comment by harubaru** [comment]
i am legitimately confused

**Comment by ArEnSc** [comment]
> i was wondering last night and i may be wrong here but could the training data lets say twitter be encoded such that a base52 scheme is used what do i mean by this well lets take an example "word" which occurs approximately 92.31% of the way through an English dictionary which could contain up to 1.25M words including rare scientific words etc this means "word" in our lookup table is 1153875 base52 encoded Here is the complete base 52 encoding of 1153875: 1153875 / 52 = 22074 R 25 22074 / 52 = 423 R 24 423 / 52 = 8 R 15 8 / 52 = 0 R 8 Thus, the base 52 encoding of 1153875 is zzzy.
> 
> which saves nothing but if every word becomes 4 letters, you have a new language, basically symbolic representations for the bot to use for its thoughts not sure if this is any use or not?

I think you are coming up with a odd way to manually create an embedding layer or positional encoding? When sentences are tokenized using BPE or other methods the become numbers, those numbers are then learned by the embedding layer and relationships of those numbers are built "Like" but not the same as word vectors but if you take a look at the article below it should make sense

https://towardsdatascience.com/word-vectors-and-word-meaning-90493d13af76#:~:text=WHAT%20IS%20A%20WORD%20VECTOR,frequencies%20are%20represented%20with%20numbers.

**Comment by scatterp2** [comment]
thanks for the responses 
@harubaru Sorry I suck at explaining myself but ArEnSc got my idea.. 
my thoughts were exactly on what ArEnSc said "become numbers" I was just pondering on if that happens in the most efficient way.
I.E is the symbolic number representing the word using the least amount of bytes possible

**Comment by ArEnSc** [comment]
> thanks for the responses @harubaru Sorry I suck at explaining myself but ArEnSc got my idea.. my thoughts were exactly on what ArEnSc said "become numbers" I was just pondering on if that happens in the most efficient way. I.E is the symbolic number representing the word using the least amount of bytes possible

yeah it's already done as efficiently as possible as far as I know, I am not sure what your situation is in AI and chat bot understanding but these are the resources that have helped me, maybe they can help you develop a stronger intuition 
https://jalammar.github.io/illustrated-transformer/

**Comment by harubaru** [comment]
> > It seems a bit controversial as to whether or not OPT beats Neo or Neo beats OPT, at least from some places I've asked. I've used Neo as the base for 125M, 1.3B, and 2.7B since people recommended to me that those were good choices over OPT. I'm planning on using OPT 13B and above though
> 
> That's a good idea. Is there a way to get a normal pytorch_model.bin instead of safetensor?

@Delcos I've done some digging, OPT underperforms GPT Neo and GPT Neo X. I'm going to be using Pythia 13B (deduped) and GPT Neo X 20B for the later models.

**Comment by harubaru** [comment]
| Model | LAMBADA Accuracy |
|--------------|--------------|
|OPT 13B | 70.23% |
|Pythia 13B | 70.46% |
|Pythia 13B Deduped | 71.01% |
